crates/bevy_math/src/curves/fixed.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    curves::{Curve, CurveCursor},
+    interpolation::Lerp,
+};
+
+// TODO: impl Serialize, Deserialize
+/// Curve with evenly spaced keyframes, in another words a curve with a fixed frame rate.
+///
+/// This curve maintains the faster sampling rate over a wide range of frame rates, because
+/// it doesn't rely on keyframe cursor. As a downside, it will have a bigger memory foot print.
+#[derive(Default, Debug)]
+pub struct CurveFixed<T> {
+    frame_rate: f32,
+    /// Negative number of frames before the curve starts
+    negative_offset: f32,
```

Comment:
The number of frames is an f32? That seems surprising to me.

Reply:
And yet this value is passed in and out as an integer. Hmm...

Reply:
integer to float conversion and vise versa are hot instructions, I always try to put the data in the most convenient format possible for the sample function;

Plus it's possible to have fractional frame rates like the 29.97 that you find in some videos;

Reply:
Addressed in the new update